pkg/apis/migration/v1alpha1/condition.go

```diff
@@ -17,16 +17,23 @@ const (
 )
 
 // Condition
+// Type - The condition type.
+// Status - The condition status.
+// Reason - The reason for the condition.
+// Message - The human readable description of the condition.
+// staged - The condition is staged for commit.
 type Condition struct {
 	Type               string      `json:"type"`
 	Status             string      `json:"status"`
 	Reason             string      `json:"reason,omitempty"`
 	Message            string      `json:"message,omitempty"`
 	LastTransitionTime metav1.Time `json:"lastTransitionTime"`
+	staged             bool
 }
 
 // Update this condition with another's fields.
 func (r *Condition) Update(other Condition) {
```

Comment:
I'm trying to understand the intent here based on the comment on this method. 
```
Update this condition with another's fields.
```
Why would you want to update a condition with another conditions fields?

Reply:
This is mainly used by `SetCondition()` when a condition _Type_ already exists.  The _other_ condition is a new one passed to SetCondition().  This updates an existing condition with a new one excluding the LastTransitionTime.  It supports SetCondition() being idempotent while preserving the LastTransitionTime.